play.pokemonshowdown.com/src/client-main.ts

```diff
@@ -373,6 +387,69 @@ class PSTeams extends PSStreamModel<'team' | 'format'> {
 			key: '',
 		};
 	}
+	loadRemoteTeams() {
+		PSLoginServer.query('getteams', {}).then(data => {
+			if (!data) return;
+			if (data.actionerror) {
+				return PS.alert('Error loading uploaded teams: ' + data.actionerror);
+			}
+			for (const team of data.teams as ((Team & { team: string })[])) {
+				let matched = false;
+				for (const curTeam of this.list) {
```

Comment:
O(n^2)? oof. You could bring it down to O(n) after sorting both lists.

Reply:
I ended up not doing this. The code got too complex, and with my other optimizations, I think it's unnecessary.